rosidl_generator_c/rosidl_generator_c/__init__.py

```diff
@@ -166,7 +166,7 @@ def primitive_value_to_c(type_, value):
     if type_ == 'bool':
         return 'true' if value else 'false'
 
-    if type_ in ['byte', 'char', 'int8', 'int16', 'int32', 'int64']:
+    if type_ in ['byte', 'char', 'signed char', 'int8', 'int16', 'int32', 'int64']:
```

Comment:
`signed char` is not a valid type name. Valid type names are only the keys of the `MSG_TYPE_TO_C` dict.